src/Cli/dotnet/CommonOptions.cs

```diff
@@ -174,6 +174,12 @@ internal static string ArchOptionValue(ParseResult parseResult) =>
             // Flip the argument so that if this option is specified we get selfcontained=false
             .SetForwardingFunction((arg, p) => ForwardSelfContainedOptions(!arg, p));
 
+        public static readonly CliOption<IEnumerable<string>> EnvOption = new CliOption<IEnumerable<string>>("--environment", "-e")
```

Comment:
One thing to consider here is to make the type of the Option a `Dictionary<string, string>` directly - we do this in the containerize tool in the [CLI already](https://github.com/dotnet/sdk/blob/0ee30b8bee52e1104e5532645397a7450a226cde/src/Containers/containerize/ContainerizeCommand.cs#L300-L319).  This ensures that people that read the option don't have to re-implement parsing the key/value dictionaries.